### PR TITLE
Per-pool project limits (from Incus)

### PR DESCRIPTION
**Pull request by MggMuggins**
Implements `limits.disk.pool.POOLNAME` config key on projects. When set, the key prevents a project using more than the configured amount of space within that pool.

When `limits.disk.pool.POOLNAME: 0`, the pool is excluded from `lxc storage list` in that project.

Contains cherry-picks from https://github.com/lxc/incus/pull/1098

### CODE REVIEW

File: test/suites/projects.sh [hunk 1]
```diff
@@ -561,6 +561,43 @@ test_projects_limits() {
 
   deps/import-busybox --project p1 --alias testimage
 
+  # Test per-pool limits.
```
Comment (tomponline) [inline test/suites/projects.sh:564]: Please can we add tests for the hidden pools feature too.

File: doc/api-extensions.md [hunk 2]
```diff
@@ -2464,3 +2464,7 @@ Adds the following internal metrics:
 
 * Total completed requests
 * Number of ongoing requests
+
+## `projects_limits_disk_pool`
+
+This introduces per-pool project disk limits, introducing a `limits.disk.pool.NAME` configuration option to the project limits.
```
Comment (tomponline) [inline doc/api-extensions.md:2470]: Please can this mention this bit:

> When limits.disk.pool.POOLNAME: 0, the pool is excluded from storage pool list in that project.

File: lxd/api_project.go [hunk 3]
```diff
@@ -1380,6 +1380,34 @@ func projectValidateConfig(s *state.State, config map[string]string) error {
...
+
+		// Load all the pools.
+		pools, err := tx.GetStoragePoolNames(ctx)
+		if err != nil {
+			return err
+		}
+
+		// Add the storage-pool specific config keys.
+		for _, poolName := range pools {
+			// lxdmeta:generate(entity=project, group=limits, key=limits.disk.pool.POOL_NAME)
```
Comment (tomponline) [inline lxd/api_project.go:1395]: Please can this mention:

> > When limits.disk.pool.POOLNAME: 0, the pool is excluded from storage pool list in that project.

File: test/suites/projects.sh (hunk 1 above)
Comment (MggMuggins) [reply test/suites/projects.sh:564]: This is checked on lines 572-573

**Review by tomponline — APPROVED** [review]
Ta!

**Comment by MusicDin** [comment]
Bit late, but should `limits.disk.pool.<storage-pool>` option be documented in [Project Limits](https://documentation.ubuntu.com/lxd/en/latest/reference/projects/#project-limits)?

Maybe at least under `limits.disk` if a separate option is not desired?

**Comment by tomponline** [comment]
Good spot @MusicDin !

@MggMuggins please can you open a follow up PR, thanks.

**Pull request merged**